devito/passes/iet/misc.py

```diff
@@ -146,5 +147,11 @@ def is_on_device(maybe_symbol, gpu_fit, only_writes=False):
             expressions = FindNodes(Expression).visit(iet)
             functions &= {i.write for i in expressions}
 
+    fsave = [f for f in functions if f.is_TimeFunction and f.save is not None]
+    if 'all-fallback' in gpu_fit and fsave:
+        warning("gpu-fit option didn't include save-full TimeFunctions %s which "
```

Comment:
nitpicking: maybe something "easier" such as "TimeFunction %s assumed to fit the GPU memory" ?